x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlConfiguration.java

```diff
@@ -144,6 +159,36 @@ public Map<String, Map<String, Column>> tables() {
         return tables;
     }
 
+    /**
+     * Get a {@link LocalRelation} from the {@link #tables()} or {@code null}
+     * if there's no parameter with that name. This will cache the result so
+     * the same name will always return the same {@linkplain LocalRelation}.
+     * That isn't strictly needed, but it feels nice.
+     */
+    public LocalRelation localRelationFromTable(String name) {
+        LocalRelation localRelation = tablesFromLocalRelation.get(name);
+        if (localRelation != null) {
+            return localRelation;
+        }
+        Map<String, Column> table = tables.get(name);
+        if (table == null) {
+            return null;
+        }
+        Block[] blocks = new Block[table.size()];
+
+        List<Attribute> attributes = new ArrayList<>(blocks.length);
+        int i = 0;
+        for (Map.Entry<String, Column> entry : table.entrySet()) {
+            Column column = entry.getValue();
+            EsField field = new EsField(entry.getKey(), column.type(), Map.of(), false, false);
+            attributes.add(new FieldAttribute(Source.EMPTY, null, entry.getKey(), field));
```

Comment:
Thought: I just realized we act as if we had fields from ES here; maybe it would be more correct to use either reference attributes (then we don't need to reference-ify the field attrs later), or even introduce a new attribute type (local attribute or so).